From edec1e9a3c80668300fa566840a5e1b7139204d0 Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Wed, 6 Aug 2014 19:39:44 +0800
Subject: [PATCH 4746/5965] PD #93541:Remove unused variable warning

Change-Id: Iab63c7faf792fb43e94ee2c991ca56b566754948

Conflicts:

	arch/arm/mach-meson8/lcd/edp_drv.c

Conflicts:

	drivers/amlogic/amports/vh265.c
---
 Makefile                                      |  3 +-
 arch/arm/mach-meson6/cpu.c                    |  2 +-
 .../mach-meson6/hdmi_tx_hw/hdmi_tx_cec_hw.c   |  4 +-
 arch/arm/mach-meson6/lcd/lcd_config.c         | 18 +++-
 arch/arm/mach-meson6/lcd/lcd_config.h         | 11 ---
 arch/arm/mach-meson8/lcd/edp_drv.c            | 73 ++++++++++-------
 arch/arm/mach-meson8/lcd/edp_drv.h            | 13 +--
 arch/arm/mach-meson8/lcd/lcd_config.c         | 17 ++++
 arch/arm/mach-meson8/lcd/lcd_config.h         |  9 --
 arch/arm/mach-meson8/lcd/mipi_dsi_util.c      | 13 +--
 arch/arm/mach-meson8b/lcd/lcd_config.c        | 12 ++-
 arch/arm/mach-meson8b/lcd/lcd_config.h        |  9 --
 arch/arm/mach-meson8b/lcd/mipi_dsi_util.c     | 13 +--
 drivers/amlogic/amports/vh264_4k2k.c          |  2 -
 drivers/amlogic/amports/video.c               |  6 +-
 drivers/amlogic/bluetooth/rtk8723/rtk_btusb.c |  3 -
 drivers/amlogic/camera/common/vm.c            |  2 +-
 drivers/amlogic/deinterlace/deinterlace.c     |  8 +-
 drivers/amlogic/display/osd_ext/osd_hw.c      | 21 +++--
 .../display/vout/lcd_extern/i2c_tc101.c       |  7 +-
 .../display/vout/lcd_extern/lcd_extern.c      | 24 +++++-
 .../display/vout/lcd_extern/mipi_N070ICN.c    |  1 +
 .../display/vout/lcd_extern/spi_LD070WS2.c    |  9 +-
 drivers/amlogic/display/vout/lcdoutc.c        | 18 ++++
 drivers/amlogic/ethernet/am_net8218.c         |  1 -
 drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c    |  4 +-
 .../amlogic/input/new_remote/remote_func.c    | 19 +++++
 .../amlogic/input/new_remote/remote_main.c    | 62 ++++++++++++++
 .../amlogic/input/new_remote/remote_main.h    | 82 +------------------
 drivers/amlogic/input/remote/sw_remote_kbd.c  |  2 +-
 drivers/amlogic/input/saradc/saradc.c         |  1 -
 drivers/amlogic/input/sensor/lsm303d.c        |  4 +-
 drivers/amlogic/input/sensor/mir3da_cust.c    |  2 +-
 .../input/touchscreen/gslx680_compatible.c    |  2 +-
 drivers/amlogic/ionvideo/ionvideo.c           |  2 +-
 drivers/amlogic/securitykeys/core.c           |  2 +-
 drivers/amlogic/watchdog/aml_wdt.c            |  3 -
 drivers/media/dvb-core/dvb_frontend.c         |  4 +-
 include/linux/amlogic/vout/aml_lcd_common.h   | 13 +++
 include/linux/amlogic/vout/aml_lcd_extern.h   |  1 +
 include/linux/amlogic/vout/lcdoutc.h          | 26 ------
 include/linux/amlogic/vout/vout_notify.h      |  2 +-
 sound/soc/aml/m6/aml_audio_hw.c               |  2 +-
 sound/soc/aml/m6/aml_audio_hw_pcm2bt.c        |  2 +-
 sound/soc/aml/m6/aml_m6_wm8960.c              |  8 +-
 sound/soc/aml/m6/aml_pcm.c                    | 18 ++--
 sound/soc/aml/m6/aml_platform.c               |  4 +-
 sound/soc/aml/m6/aml_platform_pcm2bt.c        |  2 +-
 sound/soc/aml/m8/aml_m8.c                     |  1 -
 49 files changed, 311 insertions(+), 256 deletions(-)
 create mode 100644 include/linux/amlogic/vout/aml_lcd_common.h

diff --git a/Makefile b/Makefile
index 543998c2ff02..33b31929c08e 100755
--- a/Makefile
+++ b/Makefile
@@ -379,7 +379,8 @@ KBUILD_CFLAGS   += -Werror=enum-compare \
 		   -Werror=implicit-int \
 		   -Werror=missing-braces \
 		   -Werror=unused-value \
-		   -Werror=maybe-uninitialized
+		   -Werror=maybe-uninitialized \
+		   -Werror=unused-variable
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
 KBUILD_AFLAGS   := -D__ASSEMBLY__
diff --git a/arch/arm/mach-meson6/cpu.c b/arch/arm/mach-meson6/cpu.c
index 95276ff879c4..6fb5a011cec3 100755
--- a/arch/arm/mach-meson6/cpu.c
+++ b/arch/arm/mach-meson6/cpu.c
@@ -29,7 +29,7 @@
 static int meson_cpu_version[MESON_CPU_VERSION_LVL_MAX+1];
 int __init meson_cpu_version_init(void)
 {
-	unsigned int version,ver=0xa;
+	unsigned int ver=0xa;
 	unsigned int  *version_map;
 
 	meson_cpu_version[MESON_CPU_VERSION_LVL_MAJOR] =
diff --git a/arch/arm/mach-meson6/hdmi_tx_hw/hdmi_tx_cec_hw.c b/arch/arm/mach-meson6/hdmi_tx_hw/hdmi_tx_cec_hw.c
index dae283e44b4f..8b4a112339eb 100755
--- a/arch/arm/mach-meson6/hdmi_tx_hw/hdmi_tx_cec_hw.c
+++ b/arch/arm/mach-meson6/hdmi_tx_hw/hdmi_tx_cec_hw.c
@@ -135,7 +135,7 @@ int cec_ll_tx_polling(const unsigned char *msg, unsigned char len)
     int i;
     unsigned int ret = 0xf;
     unsigned int n;
-	unsigned int j = 30;
+//	unsigned int j = 30;
     int pos;
 
     for (i = 0; i < len; i++) {
@@ -401,4 +401,4 @@ void cec_gpi_init(void)
 void cec_test_(unsigned int cmd)
 {
     printk("CEC: bus level: %s\n", BUS_LEVEL() ? "High" : "Low");
-}
\ No newline at end of file
+}
diff --git a/arch/arm/mach-meson6/lcd/lcd_config.c b/arch/arm/mach-meson6/lcd/lcd_config.c
index 9c7fe749c309..88e2e1f8a056 100755
--- a/arch/arm/mach-meson6/lcd/lcd_config.c
+++ b/arch/arm/mach-meson6/lcd/lcd_config.c
@@ -35,6 +35,7 @@
 #include <mach/am_regs.h>
 #include <mach/lcd_reg.h>
 #include <linux/amlogic/vout/lcdoutc.h>
+#include <linux/amlogic/vout/aml_lcd_common.h>
 #include <mach/clock.h>
 #include <mach/mod_gate.h>
 #include <asm/fiq.h>
@@ -54,6 +55,17 @@ static unsigned gamma_cntl_port_offset = 0;
 
 void lcd_config_init(Lcd_Config_t *pConf);
 
+#define SS_LEVEL_MAX	7
+static const char *lcd_ss_level_table[]={
+	"0",
+	"0.5%",
+	"1%",
+	"2%",
+	"3%",
+	"4%",
+	"5%",
+};
+
 static void print_lcd_driver_version(void)
 {
     printk("lcd driver version: %s%s\n\n", LCD_DRV_DATE, LCD_DRV_TYPE);
@@ -63,12 +75,10 @@ static void lcd_ports_ctrl_lvds(Bool_t status)
 {
 	if (status) {
 		WRITE_LCD_REG_BITS(LVDS_GEN_CNTL, 1, 3, 1); //enable lvds fifo
-		if (lcd_Conf->lcd_basic.lcd_bits == 6) {
+		if (lcd_Conf->lcd_basic.lcd_bits == 6)
 			WRITE_LCD_REG_BITS(LVDS_PHY_CNTL4, 0x27, 0, 7);	//enable LVDS 3 channels
-		}
-		else {
+		else
 			WRITE_LCD_REG_BITS(LVDS_PHY_CNTL4, 0x2f, 0, 7); //enable LVDS 4 channels
-		}
 	}
 	else {
 		WRITE_LCD_REG_BITS(LVDS_PHY_CNTL3, 0, 0, 1);
diff --git a/arch/arm/mach-meson6/lcd/lcd_config.h b/arch/arm/mach-meson6/lcd/lcd_config.h
index 534e7b969bb2..b38f18baf574 100755
--- a/arch/arm/mach-meson6/lcd/lcd_config.h
+++ b/arch/arm/mach-meson6/lcd/lcd_config.h
@@ -175,15 +175,4 @@ static const unsigned od_table[4] = {1,2,4,8};
 static const unsigned div_pre_table[6] = {1,2,3,4,5,6};
 //********************************************//
 
-#define SS_LEVEL_MAX	7
-static const char *lcd_ss_level_table[]={
-	"0",
-	"0.5%",
-	"1%",
-	"2%",
-	"3%",
-	"4%",
-	"5%",
-};
-
 #endif
diff --git a/arch/arm/mach-meson8/lcd/edp_drv.c b/arch/arm/mach-meson8/lcd/edp_drv.c
index d17bd59e17fd..36cdc708a47f 100755
--- a/arch/arm/mach-meson8/lcd/edp_drv.c
+++ b/arch/arm/mach-meson8/lcd/edp_drv.c
@@ -28,6 +28,25 @@ static struct class *edp_debug_class = NULL;
 // dplpm for eDP link policy maker
 //*************************************//
 
+typedef enum {
+	dpcd_strings_0 = 0,
+	dpcd_strings_1,
+	dpcd_strings_2,
+	dpcd_strings_3,
+	dpcd_strings_4,
+	dpcd_strings_5,
+	dpcd_strings_6,
+	dpcd_strings_7,
+	dpcd_strings_8,
+	dpcd_strings_9,
+	dpcd_strings_10,
+	dpcd_strings_11,
+	dpcd_strings_12,
+	dpcd_strings_13,
+	dpcd_strings_14,
+	dpcd_strings_15,
+} EDP_DPCD_Strings_t;
+
 // Strings used by printout functions
 static const char *dpcd_strings[] =
 {
@@ -54,7 +73,7 @@ static EDP_Link_Config_t lconfig = {
 	.max_link_rate = VAL_EDP_TX_LINK_BW_SET_270,
 	.enhanced_framing_en = 0,
 	.lane_count = 4,
-	.link_rate = VAL_EDP_TX_LINK_BW_SET_270,	
+	.link_rate = VAL_EDP_TX_LINK_BW_SET_270,
 	.vswing = VAL_EDP_TX_PHY_VSWING_0,
 	.preemphasis = VAL_EDP_TX_PHY_PREEMPHASIS_0,
 	.ss_level = 0,
@@ -167,7 +186,7 @@ static void dptx_dump_MSA(void)
            READ_DPTX_REG(EDP_TX_MAIN_STREAM_DATA_COUNT_PER_LANE));
     DPRINT("********************************************\n");
 }
-
+#if 0
 static int dptx_set_link_rate(unsigned char link_rate)
 {
 	int status = 0;
@@ -217,7 +236,7 @@ static int dptx_set_lane_count(unsigned char lane_count)
 	
 	return status;
 }
-
+#endif
 //Main Stream Attributes
 static void dptx_set_MSA(EDP_MSA_t *vm)
 {
@@ -651,44 +670,44 @@ static int trdp_get_sink_caps(TRDP_DPCDData_t *dpcd_data)
         return status;
 
     // Clear data structures
-    memset(dpcd_data->downstream_port_types, 0, 16);
-    memset(dpcd_data->downstream_port_caps, 0, 16);
+    memset(dpcd_data->downstream_port_types, 0, 10);
+    memset(dpcd_data->downstream_port_caps, 0, 10);
 
     switch (aux_data[0]) {
         case 0x10:
             dpcd_data->dpcd_rev = 0x10;
-            dpcd_data->rev_string = dpcd_strings[5];
+            dpcd_data->rev_string = dpcd_strings_5;
             break;
         case 0x11:
             dpcd_data->dpcd_rev = 0x11;
-            dpcd_data->rev_string = dpcd_strings[6];
+            dpcd_data->rev_string = dpcd_strings_6;
             break;
         case 0x12:
             dpcd_data->dpcd_rev = 0x12;
-            dpcd_data->rev_string = dpcd_strings[14];
+            dpcd_data->rev_string = dpcd_strings_14;
             break;
         default:
             dpcd_data->dpcd_rev = 0x00;
-            dpcd_data->rev_string = dpcd_strings[0];
+            dpcd_data->rev_string = dpcd_strings_0;
             break;
     }
 
     switch (aux_data[1]) {
         case VAL_EDP_TX_LINK_BW_SET_162:
             dpcd_data->max_link_rate = VAL_EDP_TX_LINK_BW_SET_162;
-            dpcd_data->link_rate_string = dpcd_strings[7];
+            dpcd_data->link_rate_string = dpcd_strings_7;
             break;
         case VAL_EDP_TX_LINK_BW_SET_270:
             dpcd_data->max_link_rate = VAL_EDP_TX_LINK_BW_SET_270;
-            dpcd_data->link_rate_string = dpcd_strings[8];
+            dpcd_data->link_rate_string = dpcd_strings_8;
             break;
         case VAL_EDP_TX_LINK_BW_SET_540:
             dpcd_data->max_link_rate = VAL_EDP_TX_LINK_BW_SET_540;
-            dpcd_data->link_rate_string = dpcd_strings[15];
+            dpcd_data->link_rate_string = dpcd_strings_15;
             break;
         default:
             dpcd_data->max_link_rate = 0;
-            dpcd_data->link_rate_string = dpcd_strings[1];
+            dpcd_data->link_rate_string = dpcd_strings_1;
             break;
     }
 
@@ -725,22 +744,22 @@ static int trdp_get_sink_caps(TRDP_DPCDData_t *dpcd_data)
             // Downstream port type
             switch ((dpcd_data->downstream_port_caps[xx] & 0x07)) {
                 case 0x01:
-                    dpcd_data->downstream_port_types[xx] = dpcd_strings[9];
+                    dpcd_data->downstream_port_types[xx] = dpcd_strings_9;
                     break;
                 case 0x02:
-                    dpcd_data->downstream_port_types[xx] = dpcd_strings[10];
+                    dpcd_data->downstream_port_types[xx] = dpcd_strings_10;
                     break;
                 case 0x03:
-                    dpcd_data->downstream_port_types[xx] = dpcd_strings[11];
+                    dpcd_data->downstream_port_types[xx] = dpcd_strings_11;
                     break;
                 case 0x04:
-                    dpcd_data->downstream_port_types[xx] = dpcd_strings[12];
+                    dpcd_data->downstream_port_types[xx] = dpcd_strings_12;
                     break;
                 case 0x05:
-                    dpcd_data->downstream_port_types[xx] = dpcd_strings[13];
+                    dpcd_data->downstream_port_types[xx] = dpcd_strings_13;
                     break;
                 default:
-                    dpcd_data->downstream_port_types[xx] = dpcd_strings[1];
+                    dpcd_data->downstream_port_types[xx] = dpcd_strings_1;
                     break;
             }
         }
@@ -780,8 +799,8 @@ static int trdp_dump_DPCD(void)
            "     Has EDID                    : %s\n"
            "     Uses Previous Port          : %s\n"
            "     Buffer Size                 : %u\n",
-            dpcdinfo.rev_string, 
-            dpcdinfo.link_rate_string, 
+            dpcd_strings[dpcdinfo.rev_string], 
+            dpcd_strings[dpcdinfo.link_rate_string], 
             dpcdinfo.max_lane_count, 
            (dpcdinfo.enhanced_framing_support == 1) ? dpcd_strings[3] : dpcd_strings[4],
            (dpcdinfo.downspread_support == 1) ? "0.5%" : "None", 
@@ -800,7 +819,7 @@ static int trdp_dump_DPCD(void)
         DPRINT("   Downstream Port %u:\n"
                "      Port Type         : %s\n"
                "      HPD Aware         : %s\n",
-               i, dpcdinfo.downstream_port_types[i], 
+               i, dpcd_strings[dpcdinfo.downstream_port_types[i]], 
                ((dpcdinfo.downstream_port_caps[i] & 0x08) == 0x08) ? dpcd_strings[3] : dpcd_strings[4] );
     }
 	DPRINT("********************************************\n");
@@ -873,7 +892,7 @@ static void trdp_edp_link_rate_update(unsigned char link_rate)
 			break;
 	}
 }
-
+#if 0
 static int trdp_set_link_rate(unsigned char link_rate)
 {
 	int status = VAL_EDP_TX_AUX_OPERATION_SUCCESS;
@@ -905,8 +924,8 @@ static int trdp_set_lane_count(unsigned char lane_count)
 	
 	lcd_print("set lane count\n");
 	switch (lane_count) {
-		case 1:			
-		case 2:			
+		case 1:
+		case 2:
 		case 4:
 			enhance_framing_mode = (READ_DPTX_REG(EDP_TX_ENHANCED_FRAME_EN) & 0x01);
 			if (enhance_framing_mode)
@@ -921,7 +940,7 @@ static int trdp_set_lane_count(unsigned char lane_count)
 	
 	return status;
 }
-
+#endif
 static int trdp_set_downspread(unsigned char ss_enable)
 {
 	int status = 0;
@@ -1992,8 +2011,6 @@ static int remove_edp_attr(void)
 
 void edp_probe(void)
 {
-	int ret;
-	
 	creat_edp_attr();
 }
 
diff --git a/arch/arm/mach-meson8/lcd/edp_drv.h b/arch/arm/mach-meson8/lcd/edp_drv.h
index 04c28534cece..5a51a5cfc183 100755
--- a/arch/arm/mach-meson8/lcd/edp_drv.h
+++ b/arch/arm/mach-meson8/lcd/edp_drv.h
@@ -183,12 +183,7 @@ static const unsigned char edp_link_rate_table[] = {
     VAL_EDP_TX_LINK_BW_SET_540,
     VAL_EDP_TX_INVALID_VALUE,
 };
-static const char *edp_link_rate_string_table[]={
-    "1.62Gbps",
-    "2.70Gbps",
-    "5.40Gbps",
-    "invalid",
-};
+
 static const unsigned char edp_lane_count_table[] = {1, 2, 4, VAL_EDP_TX_INVALID_VALUE};
 #define LINK_RATE_TO_CAPACITY(x)    (x * 8 / 10) //8/10 coding
 static const unsigned edp_link_capacity_table[] = { //Mbps
@@ -223,9 +218,9 @@ typedef struct {
 	unsigned char max_lane_count;
 	unsigned char max_link_rate;
 	unsigned char downstream_port_caps[10];
-	char *downstream_port_types[10];
-	char *rev_string;
-	char *link_rate_string;
+	unsigned char downstream_port_types[10];
+	unsigned char rev_string;
+	unsigned char link_rate_string;
 	char *port_type_string;
 	unsigned num_rcv_ports;
 	unsigned num_downstream_ports;
diff --git a/arch/arm/mach-meson8/lcd/lcd_config.c b/arch/arm/mach-meson8/lcd/lcd_config.c
index d7040533d7cf..f40576211242 100755
--- a/arch/arm/mach-meson8/lcd/lcd_config.c
+++ b/arch/arm/mach-meson8/lcd/lcd_config.c
@@ -35,6 +35,7 @@
 #include <mach/am_regs.h>
 #include <mach/lcd_reg.h>
 #include <linux/amlogic/vout/lcdoutc.h>
+#include <linux/amlogic/vout/aml_lcd_common.h>
 #include <mach/clock.h>
 #include <mach/vpu.h>
 #include <mach/mod_gate.h>
@@ -56,6 +57,22 @@ static unsigned char lcd_gamma_init_err = 0;
 
 void lcd_config_init(Lcd_Config_t *pConf);
 
+#define SS_LEVEL_MAX	5
+static const char *lcd_ss_level_table[]={
+	"0",
+	"0.5%",
+	"1%",
+	"1.5%",
+	"2%",
+};
+
+static const char *edp_link_rate_string_table[]={
+    "1.62Gbps",
+    "2.70Gbps",
+    "5.40Gbps",
+    "invalid",
+};
+
 static void print_lcd_driver_version(void)
 {
     printk("lcd driver version: %s%s\n\n", LCD_DRV_DATE, LCD_DRV_TYPE);
diff --git a/arch/arm/mach-meson8/lcd/lcd_config.h b/arch/arm/mach-meson8/lcd/lcd_config.h
index 3b1c78effa38..83b4a0dfe597 100755
--- a/arch/arm/mach-meson8/lcd/lcd_config.h
+++ b/arch/arm/mach-meson8/lcd/lcd_config.h
@@ -187,13 +187,4 @@ static const unsigned edp_div0_table[15] = {1,2,3,4,5,7,8,9,11,13,17,19,23,29,31
 static const unsigned edp_div1_table[8] = {1,2,4,5,6,7,9,13};
 //********************************************//
 
-#define SS_LEVEL_MAX	5
-static const char *lcd_ss_level_table[]={
-	"0",
-	"0.5%",
-	"1%",
-	"1.5%",
-	"2%",
-};
-
 #endif
diff --git a/arch/arm/mach-meson8/lcd/mipi_dsi_util.c b/arch/arm/mach-meson8/lcd/mipi_dsi_util.c
index bc5279dfdc64..554bf32a01e7 100755
--- a/arch/arm/mach-meson8/lcd/mipi_dsi_util.c
+++ b/arch/arm/mach-meson8/lcd/mipi_dsi_util.c
@@ -16,7 +16,7 @@
 #include "mipi_dsi_util.h"
 
 #define DPRINT(...)		printk(__VA_ARGS__)
-
+//#define MIPI_DSI_COMMAND_READ
 //===============================================================================
 // Define MIPI DSI Default config
 //===============================================================================
@@ -237,6 +237,7 @@ static void set_mipi_dcs(int trans_type,        // 0: high speed, 1: low power
         WRITE_LCD_REG( MIPI_DSI_TOP_MEAS_CNTL, (READ_LCD_REG(MIPI_DSI_TOP_MEAS_CNTL) | (0x1<<BIT_VSYNC_MEAS_EN | (0x1<<BIT_TE_MEAS_EN))));
     }
 }
+#if 0
 // -----------------------------------------------------------------------------
 //                     Function: set_mipi_int
 // Configure relative registers for mipi interrupt
@@ -246,7 +247,7 @@ static void set_mipi_int(void)
     WRITE_LCD_REG( MIPI_DSI_DWC_INT_MSK0_OS, 0);
     WRITE_LCD_REG( MIPI_DSI_DWC_INT_MSK1_OS, 0);
 }
-
+#endif
 // ----------------------------------------------------------------------------
 // Function: wait_bta_ack
 // Poll to check if the BTA ack is finished
@@ -282,7 +283,7 @@ static void wait_cmd_fifo_empty(void)
     } while((((cmd_status >> BIT_GEN_CMD_EMPTY) & 0x1) != 0x1) && (i>0));
     print_mipi_cmd_status(i, cmd_status);
 }
-
+#ifdef MIPI_DSI_COMMAND_READ
 // ----------------------------------------------------------------------------
 // Function: wait_for_generic_read_response
 // Wait for generic read response
@@ -309,7 +310,7 @@ static unsigned int wait_for_generic_read_response(void)
     data_out = READ_LCD_REG( MIPI_DSI_DWC_GEN_PLD_DATA_OS );
     return data_out;
 }
-
+#endif
 // ----------------------------------------------------------------------------
 // Function: generic_if_wr
 // Generic interface write, address has to be MIPI_DSI_DWC_GEN_PLD_DATA_OS and
@@ -326,7 +327,7 @@ static unsigned int generic_if_wr(unsigned int address, unsigned int data_in)
 
     return 0;
 }
-
+#ifdef MIPI_DSI_COMMAND_READ
 // ----------------------------------------------------------------------------
 //                           Function: generic_if_rd
 // Generic interface read, address has to be MIPI_DSI_DWC_GEN_PLD_DATA_OS
@@ -365,7 +366,7 @@ static unsigned int generic_read_packet_0_para(unsigned char data_type, unsigned
 
     return read_data;
 }
-
+#endif
 // ----------------------------------------------------------------------------
 //                           Function: generic_write_short_packet
 // Generic Write Short Packet with Generic Interface
diff --git a/arch/arm/mach-meson8b/lcd/lcd_config.c b/arch/arm/mach-meson8b/lcd/lcd_config.c
index c3a8478ab033..738f0b90e487 100755
--- a/arch/arm/mach-meson8b/lcd/lcd_config.c
+++ b/arch/arm/mach-meson8b/lcd/lcd_config.c
@@ -35,6 +35,7 @@
 #include <mach/am_regs.h>
 #include <mach/lcd_reg.h>
 #include <linux/amlogic/vout/lcdoutc.h>
+#include <linux/amlogic/vout/aml_lcd_common.h>
 #include <mach/clock.h>
 #include <mach/vpu.h>
 #include <mach/mod_gate.h>
@@ -55,6 +56,15 @@ static unsigned char lcd_gamma_init_err = 0;
 
 void lcd_config_init(Lcd_Config_t *pConf);
 
+#define SS_LEVEL_MAX	5
+static const char *lcd_ss_level_table[]={
+	"0",
+	"0.5%",
+	"1%",
+	"1.5%",
+	"2%",
+};
+
 static void print_lcd_driver_version(void)
 {
     printk("lcd driver version: %s%s\n\n", LCD_DRV_DATE, LCD_DRV_TYPE);
@@ -996,7 +1006,7 @@ static void generate_clk_parameter(Lcd_Config_t *pConf)
     unsigned m, n, od, div_pre, div_post, xd;
     unsigned od_sel, pre_div_sel;
     unsigned div_pre_sel_max, crt_xd_max;
-    unsigned pll_vco, fout_pll, div_pre_out, div_post_out, final_freq, iflogic_vid_clk_in_max;
+    unsigned pll_vco, fout_pll, div_pre_out, div_post_out, iflogic_vid_clk_in_max;
     unsigned od_fb=0;
     unsigned int dsi_bit_rate_min=0, dsi_bit_rate_max=0;
     unsigned clk_num = 0;
diff --git a/arch/arm/mach-meson8b/lcd/lcd_config.h b/arch/arm/mach-meson8b/lcd/lcd_config.h
index b20643352697..f4ebf239155a 100755
--- a/arch/arm/mach-meson8b/lcd/lcd_config.h
+++ b/arch/arm/mach-meson8b/lcd/lcd_config.h
@@ -179,13 +179,4 @@ static const unsigned od_table[4] = {1,2,4,8};
 static const unsigned div_pre_table[6] = {1,2,3,4,5,6};
 //********************************************//
 
-#define SS_LEVEL_MAX	5
-static const char *lcd_ss_level_table[]={
-	"0",
-	"0.5%",
-	"1%",
-	"1.5%",
-	"2%",
-};
-
 #endif
diff --git a/arch/arm/mach-meson8b/lcd/mipi_dsi_util.c b/arch/arm/mach-meson8b/lcd/mipi_dsi_util.c
index bc5279dfdc64..554bf32a01e7 100755
--- a/arch/arm/mach-meson8b/lcd/mipi_dsi_util.c
+++ b/arch/arm/mach-meson8b/lcd/mipi_dsi_util.c
@@ -16,7 +16,7 @@
 #include "mipi_dsi_util.h"
 
 #define DPRINT(...)		printk(__VA_ARGS__)
-
+//#define MIPI_DSI_COMMAND_READ
 //===============================================================================
 // Define MIPI DSI Default config
 //===============================================================================
@@ -237,6 +237,7 @@ static void set_mipi_dcs(int trans_type,        // 0: high speed, 1: low power
         WRITE_LCD_REG( MIPI_DSI_TOP_MEAS_CNTL, (READ_LCD_REG(MIPI_DSI_TOP_MEAS_CNTL) | (0x1<<BIT_VSYNC_MEAS_EN | (0x1<<BIT_TE_MEAS_EN))));
     }
 }
+#if 0
 // -----------------------------------------------------------------------------
 //                     Function: set_mipi_int
 // Configure relative registers for mipi interrupt
@@ -246,7 +247,7 @@ static void set_mipi_int(void)
     WRITE_LCD_REG( MIPI_DSI_DWC_INT_MSK0_OS, 0);
     WRITE_LCD_REG( MIPI_DSI_DWC_INT_MSK1_OS, 0);
 }
-
+#endif
 // ----------------------------------------------------------------------------
 // Function: wait_bta_ack
 // Poll to check if the BTA ack is finished
@@ -282,7 +283,7 @@ static void wait_cmd_fifo_empty(void)
     } while((((cmd_status >> BIT_GEN_CMD_EMPTY) & 0x1) != 0x1) && (i>0));
     print_mipi_cmd_status(i, cmd_status);
 }
-
+#ifdef MIPI_DSI_COMMAND_READ
 // ----------------------------------------------------------------------------
 // Function: wait_for_generic_read_response
 // Wait for generic read response
@@ -309,7 +310,7 @@ static unsigned int wait_for_generic_read_response(void)
     data_out = READ_LCD_REG( MIPI_DSI_DWC_GEN_PLD_DATA_OS );
     return data_out;
 }
-
+#endif
 // ----------------------------------------------------------------------------
 // Function: generic_if_wr
 // Generic interface write, address has to be MIPI_DSI_DWC_GEN_PLD_DATA_OS and
@@ -326,7 +327,7 @@ static unsigned int generic_if_wr(unsigned int address, unsigned int data_in)
 
     return 0;
 }
-
+#ifdef MIPI_DSI_COMMAND_READ
 // ----------------------------------------------------------------------------
 //                           Function: generic_if_rd
 // Generic interface read, address has to be MIPI_DSI_DWC_GEN_PLD_DATA_OS
@@ -365,7 +366,7 @@ static unsigned int generic_read_packet_0_para(unsigned char data_type, unsigned
 
     return read_data;
 }
-
+#endif
 // ----------------------------------------------------------------------------
 //                           Function: generic_write_short_packet
 // Generic Write Short Packet with Generic Interface
diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 29b172ff6798..4153afd2cc18 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -1382,8 +1382,6 @@ extern void AbortEncodeWithVdec2(int abort);
 static int amvdec_h264_4k2k_probe(struct platform_device *pdev)
 {
     struct resource *mem;
-    int count = 0;
-
     printk("amvdec_h264_4k2k probe start.\n");
     mutex_lock(&vh264_4k2k_mutex);
     
diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index 8bc2a112bd02..1646ee38d436 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -1214,8 +1214,9 @@ u32 property_changed_true=0;
 static void vsync_toggle_frame(vframe_t *vf)
 {
     u32 first_picture = 0;
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
     unsigned long flags;
-
+#endif
     frame_count++;
     if(debug_flag& DEBUG_FLAG_PRINT_TOGGLE_FRAME){
         printk("%s()\n", __func__);
@@ -2196,9 +2197,10 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
     s32 i, vout_type;
     vframe_t *vf;
     unsigned long flags;
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
     vdin_v4l2_ops_t *vdin_ops = NULL;
     vdin_arg_t arg;
-
+#endif
 #ifdef CONFIG_AM_VIDEO_LOG
     int toggle_cnt;
 #endif
diff --git a/drivers/amlogic/bluetooth/rtk8723/rtk_btusb.c b/drivers/amlogic/bluetooth/rtk8723/rtk_btusb.c
index 879e222f5575..7466366bc812 100755
--- a/drivers/amlogic/bluetooth/rtk8723/rtk_btusb.c
+++ b/drivers/amlogic/bluetooth/rtk8723/rtk_btusb.c
@@ -69,9 +69,6 @@ static struct file_operations btfcd_file_ops_g =
 
 #define VERSION "2.11"
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 7, 1)
-static bool reset = 0;
-#endif
 
 static struct usb_driver btusb_driver;
 static struct usb_device_id btusb_table[] = {
diff --git a/drivers/amlogic/camera/common/vm.c b/drivers/amlogic/camera/common/vm.c
index 6a6d63414393..1561e8c2aba9 100755
--- a/drivers/amlogic/camera/common/vm.c
+++ b/drivers/amlogic/camera/common/vm.c
@@ -2014,7 +2014,7 @@ static int vm_driver_probe(struct platform_device *pdev)
 #ifndef CONFIG_CMA
 	char* buf_start;
 	unsigned int buf_size;
-	struct resource *mem;
+//	struct resource *mem;
     int idx;
 
 #if 0
diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 283b0a33fa1f..1fc5158ac9a6 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -127,6 +127,7 @@ static DEFINE_SPINLOCK(di_lock2);
 #ifdef ENABLE_SPIN_LOCK_ALWAYS
 static DEFINE_SPINLOCK(di_lock2);
 #define di_lock_irqfiq_save(irq_flag, fiq_flag) \
+				fiq_flag=0; \
                 spin_lock_irqsave(&di_lock2, irq_flag);
 
 #define di_unlock_irqfiq_restore(irq_flag, fiq_flag) \
@@ -3293,8 +3294,9 @@ static void config_di_mif(DI_MIF_t* di_mif, di_buf_t*di_buf)
 static void pre_de_process(void)
 {
   int chan2_field_num = 1;
+#ifdef NEW_DI_V1
   int cont_rd = 1;
-
+#endif
 #ifdef DI_DEBUG
     di_print("%s: start\n", __func__);
 #endif
@@ -4421,8 +4423,8 @@ static irqreturn_t det3d_irq(int irq, void *dev_instance)
 #endif
 static irqreturn_t de_irq(int irq, void *dev_instance)
 {
-   unsigned int data32;
 #ifndef CHECK_DI_DONE
+   unsigned int data32;
    data32 = Rd(DI_INTR_CTRL);
    //if ( (data32 & 0xf) != 0x1 ) {
    //     printk("%s: error %x\n", __func__, data32);
@@ -5988,7 +5990,6 @@ di task
 */
 static void di_unreg_process(void)
 {
-    ulong flags=0,fiq_flag=0, irq_flag2=0;
         if((di_pre_stru.unreg_req_flag||di_pre_stru.force_unreg_req_flag||di_pre_stru.disable_req_flag)&&
             (di_pre_stru.pre_de_busy==0)){
             //printk("===unreg_req_flag\n");
@@ -6009,6 +6010,7 @@ static void di_unreg_process(void)
 #else
 /* !RUN_REG_IN_IRQ*/
 
+			ulong flags=0,fiq_flag=0, irq_flag2=0;
             if(di_pre_stru.force_unreg_req_flag||di_pre_stru.disable_req_flag){
 #ifdef DI_DEBUG
                 di_print("%s: force_unreg\n", __func__);
diff --git a/drivers/amlogic/display/osd_ext/osd_hw.c b/drivers/amlogic/display/osd_ext/osd_hw.c
index d09077af59e7..1ef5209a595a 100755
--- a/drivers/amlogic/display/osd_ext/osd_hw.c
+++ b/drivers/amlogic/display/osd_ext/osd_hw.c
@@ -1154,14 +1154,15 @@ static  void  osd1_update_disp_freescale_enable(void)
 
 static void osd1_update_coef(void)
 {
+	int vf_coef_idx = 0;
+	int *vf_coef;
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 	int i;
 	int hf_coef_idx = 0;
 	int hf_coef_wren = 1;
-
-	int vf_coef_idx = 0;
 	int vf_coef_wren = 1;
-	int *hf_coef, *vf_coef;
-
+	int *hf_coef;
+#endif
 	if (vf_coef_idx == 0){
 		vf_coef = filt_coef0;
 	}else if (vf_coef_idx == 1){
@@ -1278,14 +1279,16 @@ static  void  osd2_update_disp_freescale_enable(void)
 
 static void osd2_update_coef(void)
 {
+	int vf_coef_idx = 0;
+	int  *vf_coef;
+
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 	int i;
 	int hf_coef_idx = 0;
-	int hf_coef_wren = 1;
-
-	int vf_coef_idx = 0;
 	int vf_coef_wren = 1;
-	int *hf_coef, *vf_coef;
-
+	int hf_coef_wren = 1;
+	int *hf_coef;
+#endif
 	if (vf_coef_idx == 0){
 		vf_coef = filt_coef0;
 	}else if (vf_coef_idx == 1){
diff --git a/drivers/amlogic/display/vout/lcd_extern/i2c_tc101.c b/drivers/amlogic/display/vout/lcd_extern/i2c_tc101.c
index 80335689fe46..fe39a4e1c8e0 100755
--- a/drivers/amlogic/display/vout/lcd_extern/i2c_tc101.c
+++ b/drivers/amlogic/display/vout/lcd_extern/i2c_tc101.c
@@ -68,7 +68,7 @@ static int aml_i2c_write(struct i2c_client *i2client,unsigned char *buff, unsign
     
     return res;
 }
-
+#if 0
 static int aml_i2c_read(struct i2c_client *i2client,unsigned char *buff, unsigned len)
 {
     int res = 0;
@@ -93,7 +93,7 @@ static int aml_i2c_read(struct i2c_client *i2client,unsigned char *buff, unsigne
 
     return res;
 }
-
+#endif
 static int i2c_reg_read(unsigned char reg, unsigned char *buf)
 {
     int ret=0;
@@ -250,7 +250,7 @@ static int aml_tc101_probe(struct platform_device *pdev)
     if (!aml_tc101_i2c_client) {
         ret = i2c_add_driver(&aml_tc101_i2c_driver);
         if (ret) {
-            printk("[error] %s probe: add i2c_driver failed\n" LCD_EXTERN_NAME);
+            printk("[error] lcd_extern probe: add i2c_driver failed\n");
             goto lcd_extern_probe_failed;
         }
     }
@@ -266,6 +266,7 @@ lcd_extern_probe_failed:
 
 static int aml_tc101_remove(struct platform_device *pdev)
 {
+    remove_lcd_extern(lcd_ext_config);
     if (pdev->dev.platform_data)
         kfree (pdev->dev.platform_data);
     return 0;
diff --git a/drivers/amlogic/display/vout/lcd_extern/lcd_extern.c b/drivers/amlogic/display/vout/lcd_extern/lcd_extern.c
index 5c94626666cb..9ddcf6087dcb 100755
--- a/drivers/amlogic/display/vout/lcd_extern/lcd_extern.c
+++ b/drivers/amlogic/display/vout/lcd_extern/lcd_extern.c
@@ -53,18 +53,27 @@ int lcd_extern_driver_check(void)
 	return 0;
 }
 
+#define BL_EXT_NAME_LEN_MAX		50
 int get_lcd_extern_dt_data(struct device_node* of_node, struct lcd_extern_config_t *pdata)
 {
 	int err;
 	int val;
 	const char *str;
 	
-	err = of_property_read_string(of_node, "dev_name", &pdata->name);
+	err = of_property_read_string(of_node, "dev_name", &str);
 	if (err) {
-		pdata->name = "aml_lcd_extern";
+		str = "aml_lcd_extern";
 		printk("warning: get dev_name failed\n");
 	}
-	
+	pdata->name = (char *)kmalloc(sizeof(char)*BL_EXT_NAME_LEN_MAX, GFP_KERNEL);
+	if (pdata->name == NULL) {
+		printk("[get_lcd_extern_dt_data]: Not enough memory\n");
+	}
+	else {
+		memset(pdata->name, 0, BL_EXT_NAME_LEN_MAX);
+		strcpy(pdata->name, str);
+		printk("load bl_extern in dtb: %s\n", pdata->name);
+	}
 	err = of_property_read_u32(of_node, "type", &pdata->type);
 	if (err) {
 		pdata->type = LCD_EXTERN_MAX;
@@ -168,3 +177,12 @@ int get_lcd_extern_dt_data(struct device_node* of_node, struct lcd_extern_config
 	
 	return 0;
 }
+
+int remove_lcd_extern(struct lcd_extern_config_t *pdata)
+{
+	if (pdata->name)
+		kfree(pdata->name);
+		
+	return 0;
+}
+
diff --git a/drivers/amlogic/display/vout/lcd_extern/mipi_N070ICN.c b/drivers/amlogic/display/vout/lcd_extern/mipi_N070ICN.c
index 17f5984dafc8..a8ae633c2cae 100755
--- a/drivers/amlogic/display/vout/lcd_extern/mipi_N070ICN.c
+++ b/drivers/amlogic/display/vout/lcd_extern/mipi_N070ICN.c
@@ -262,6 +262,7 @@ lcd_extern_probe_failed:
 
 static int aml_N070ICN_remove(struct platform_device *pdev)
 {
+    remove_lcd_extern(lcd_ext_config);
     if (pdev->dev.platform_data)
         kfree (pdev->dev.platform_data);
     return 0;
diff --git a/drivers/amlogic/display/vout/lcd_extern/spi_LD070WS2.c b/drivers/amlogic/display/vout/lcd_extern/spi_LD070WS2.c
index 3b4b45061f3b..00b5d7324bdf 100755
--- a/drivers/amlogic/display/vout/lcd_extern/spi_LD070WS2.c
+++ b/drivers/amlogic/display/vout/lcd_extern/spi_LD070WS2.c
@@ -123,7 +123,7 @@ static void spi_write_8(unsigned char addr, unsigned char data)
     udelay(SPI_DELAY);
 }
 
-static void lcd_extern_spi_init(void)
+static int lcd_extern_spi_init(void)
 {
     int ending_flag = 0;
     int i=0;
@@ -143,9 +143,11 @@ static void lcd_extern_spi_init(void)
         i++;
     }
     printk("%s\n", __FUNCTION__);
+
+    return 0;
 }
 
-static void lcd_extern_spi_off(void)
+static int lcd_extern_spi_off(void)
 {
     int ending_flag = 0;
     int i=0;
@@ -167,6 +169,8 @@ static void lcd_extern_spi_off(void)
     printk("%s\n", __FUNCTION__);
     mdelay(10);
     spi_gpio_off();
+
+    return 0;
 }
 
 static int lcd_extern_driver_update(void)
@@ -220,6 +224,7 @@ lcd_extern_probe_failed:
 
 static int aml_LD070WS2_remove(struct platform_device *pdev)
 {
+    remove_lcd_extern(lcd_ext_config);
     if (pdev->dev.platform_data)
         kfree (pdev->dev.platform_data);
     return 0;
diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index 8db1b6c1d2ea..7d20f0d3b471 100755
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -38,6 +38,7 @@
 #include <mach/lcd_reg.h>
 #include <mach/lcdoutc.h>
 #include <linux/amlogic/vout/lcdoutc.h>
+#include <linux/amlogic/vout/aml_lcd_common.h>
 #include <linux/amlogic/vout/lcd_aml.h>
 #include <mach/clock.h>
 #include <asm/fiq.h>
@@ -69,6 +70,23 @@ void lcd_print(const char *fmt, ...)
 	va_end(args);
 }
 
+static const char* lcd_power_type_table[]={
+	"cpu",
+	"pmu",
+	"signal",
+	"init",
+	"null",
+};
+
+static const char* lcd_power_pmu_gpio_table[]={
+	"GPIO0",
+	"GPIO1",
+	"GPIO2",
+	"GPIO3",
+	"GPIO4",
+	"null",
+}; 
+
 typedef struct {
 	Lcd_Config_t *pConf;
 	vinfo_t lcd_info;
diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 99c3b5152d57..358924362f26 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -908,7 +908,6 @@ static int aml_mac_init(struct net_device *ndev)
 	data_dump(ndev->dev_addr, 6);
 #ifdef CONFIG_AML_NAND_KEY
 	int ret;
-	int use_nand_mac=0;
 	u8 mac[ETH_ALEN];
 	char *endp;
 	int j;
diff --git a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c
index d58568e407bf..5b47f9f893fe 100755
--- a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c
+++ b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_cec.c
@@ -649,10 +649,10 @@ unsigned char check_cec_msg_valid(const cec_rx_message_t* pcec_message)
 static irqreturn_t cec_isr_handler(int irq, void *dev_instance)
 {
     unsigned char rx_msg[MAX_MSG], rx_len;
-    unsigned int intr_stat = 0;
 
     //cec_disable_irq();
 #if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
+    unsigned int intr_stat = 0;
     intr_stat = aml_read_reg32(P_AO_CEC_INTR_STAT);
     hdmi_print(INF, CEC "aocec irq %x\n", intr_stat);
 
@@ -1743,8 +1743,10 @@ void cec_usrcmd_set_dispatch(const char * buf, size_t count)
     int j = 0;
     int bool = 0;
     char param[32] = {0};
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
     unsigned bit_set;
     unsigned time_set;
+#endif
     unsigned char msg[4] = {0};
     
     hdmi_print(INF, CEC "cec usrcmd set dispatch start:\n");
diff --git a/drivers/amlogic/input/new_remote/remote_func.c b/drivers/amlogic/input/new_remote/remote_func.c
index 0a02f6b0ad13..fb75eba83d6d 100755
--- a/drivers/amlogic/input/new_remote/remote_func.c
+++ b/drivers/amlogic/input/new_remote/remote_func.c
@@ -20,6 +20,25 @@
 extern int cec_power_flag;
 unsigned char cec_repeat = 10;
 #endif
+static const remotereg_t *remoteregsTab[] = {
+	RDECODEMODE_NEC,
+	RDECODEMODE_DUOKAN,
+	RDECODEMODE_MITSUBISHI,
+	RDECODEMODE_THOMSON,
+	RDECODEMODE_TOSHIBA,
+	RDECODEMODE_SONYSIRC,
+	RDECODEMODE_RC5,
+	RDECODEMODE_RESERVED,
+	RDECODEMODE_RC6,
+	RDECODEMODE_RCMM,
+	RDECODEMODE_COMCAST,
+	RDECODEMODE_SANYO,
+	RDECODEMODE_SKIPLEADER,
+	RDECODEMODE_SW,
+	RDECODEMODE_SW_NEC,
+	NULL,
+	RDECODEMODE_SW_DUOKAN
+};
 extern char *remote_log_buf;
 static int auto_repeat_count,repeat_count = 0;
 static void remote_rel_timer_sr(unsigned long data);
diff --git a/drivers/amlogic/input/new_remote/remote_main.c b/drivers/amlogic/input/new_remote/remote_main.c
index f4b8f53b334f..36024daa440e 100755
--- a/drivers/amlogic/input/new_remote/remote_main.c
+++ b/drivers/amlogic/input/new_remote/remote_main.c
@@ -57,6 +57,9 @@
 static struct early_suspend early_suspend;
 #endif
 
+static bool key_pointer_switch = true;
+static unsigned int FN_KEY_SCANCODE = 0x3ff;
+static unsigned int OK_KEY_SCANCODE = 0x3ff;
 type_printk input_dbg;
 static DEFINE_MUTEX(remote_enable_mutex);
 static DEFINE_MUTEX(remote_file_mutex);
@@ -70,6 +73,65 @@ static struct remote *gp_remote = NULL;
 char *remote_log_buf;
 // use 20 map for this driver
 static __u16 key_map[20][512];
+static  irqreturn_t (*remote_bridge_sw_isr[])(int irq, void *dev_id)={
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	remote_bridge_isr,
+	remote_bridge_isr,
+};
+
+static  int (*remote_report_key[])(struct remote *remote_data)={
+	remote_hw_reprot_key,
+	remote_hw_reprot_key,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	remote_sw_reprot_key		
+};
+
+static  void (*remote_report_release_key[])(struct remote *remote_data)={
+	remote_nec_report_release_key,
+	remote_duokan_report_release_key,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	remote_sw_reprot_release_key
+};
 static __u16 mouse_map[20][6];
 int remote_printk(const char *fmt, ...)
 {
diff --git a/drivers/amlogic/input/new_remote/remote_main.h b/drivers/amlogic/input/new_remote/remote_main.h
index d578fd3d9449..c1864e810f66 100755
--- a/drivers/amlogic/input/new_remote/remote_main.h
+++ b/drivers/amlogic/input/new_remote/remote_main.h
@@ -295,25 +295,7 @@ static const remotereg_t RDECODEMODE_SANYO[] = {
 };
 
 
-static const remotereg_t *remoteregsTab[] = {
-	RDECODEMODE_NEC,
-	RDECODEMODE_DUOKAN,
-	RDECODEMODE_MITSUBISHI,
-	RDECODEMODE_THOMSON,
-	RDECODEMODE_TOSHIBA,
-	RDECODEMODE_SONYSIRC,
-	RDECODEMODE_RC5,
-	RDECODEMODE_RESERVED,
-	RDECODEMODE_RC6,
-	RDECODEMODE_RCMM,
-	RDECODEMODE_COMCAST,
-	RDECODEMODE_SANYO,
-	RDECODEMODE_SKIPLEADER,
-	RDECODEMODE_SW,
-	RDECODEMODE_SW_NEC,
-	NULL,
-	RDECODEMODE_SW_DUOKAN
-};
+
 
 extern unsigned int g_remote_base;
 #define am_remote_write_reg(x,val) aml_write_reg32(g_remote_base +x ,val)
@@ -325,9 +307,6 @@ extern unsigned int g_remote_base;
 #define am_remote_clear_mask(x,val) aml_clr_reg32_mask(g_remote_base +x,val)
 void setremotereg(const remotereg_t *r);
 
-static bool key_pointer_switch = true;
-static unsigned int FN_KEY_SCANCODE = 0x3ff;
-static unsigned int OK_KEY_SCANCODE = 0x3ff;
 
 //remote config  ioctl  cmd
 #define REMOTE_IOC_INFCODE_CONFIG	    _IOW_BAD('I',13,sizeof(short))
@@ -492,65 +471,6 @@ extern int register_fiq_bridge_handle(bridge_item_t * c_item);
 extern int unregister_fiq_bridge_handle(bridge_item_t * c_item);
 extern int fiq_bridge_pulse_trigger(bridge_item_t * c_item);
 
-static  irqreturn_t (*remote_bridge_sw_isr[])(int irq, void *dev_id)={
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	remote_bridge_isr,
-	remote_bridge_isr,
-};
-
-static  int (*remote_report_key[])(struct remote *remote_data)={
-	remote_hw_reprot_key,
-	remote_hw_reprot_key,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	remote_sw_reprot_key		
-};
-
-static  void (*remote_report_release_key[])(struct remote *remote_data)={
-	remote_nec_report_release_key,
-	remote_duokan_report_release_key,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	NULL,
-	remote_sw_reprot_release_key
-};
 
 
 #endif //_REMOTE_H
diff --git a/drivers/amlogic/input/remote/sw_remote_kbd.c b/drivers/amlogic/input/remote/sw_remote_kbd.c
index cf6f860587bf..6abfdbd15533 100755
--- a/drivers/amlogic/input/remote/sw_remote_kbd.c
+++ b/drivers/amlogic/input/remote/sw_remote_kbd.c
@@ -99,7 +99,7 @@ static int get_pulse_width(unsigned long data)
 
 static inline void kbd_software_mode_remote_wait(unsigned long data)
 {
-	unsigned short pulse_width;
+//	unsigned short pulse_width;
 	struct remote *remote_data = (struct remote *)data;
 
 //	pulse_width = get_pulse_width(data);
diff --git a/drivers/amlogic/input/saradc/saradc.c b/drivers/amlogic/input/saradc/saradc.c
index ecd07aa7e5c9..669941d94398 100755
--- a/drivers/amlogic/input/saradc/saradc.c
+++ b/drivers/amlogic/input/saradc/saradc.c
@@ -306,7 +306,6 @@ static ssize_t saradc_print_flag_store(struct class *cla, struct class_attribute
 #ifndef CONFIG_MESON_CPU_TEMP_SENSOR
 static ssize_t saradc_temperature_store(struct class *cla, struct class_attribute *attr, const char *buf, size_t count)
 {
-    int ret;
 		u8 tempsen;
 		sscanf(buf, "%d", (int*)&tempsen);
 		if (tempsen) {
diff --git a/drivers/amlogic/input/sensor/lsm303d.c b/drivers/amlogic/input/sensor/lsm303d.c
index afcddba9571b..fbaa30f93775 100644
--- a/drivers/amlogic/input/sensor/lsm303d.c
+++ b/drivers/amlogic/input/sensor/lsm303d.c
@@ -2451,7 +2451,7 @@ static ssize_t attr_get_interrupt_polarity(struct device *dev,
 	val = atomic_read(&stat->interrupt->interrupt_polarity.enable);
 	return sprintf(buf, "%d\n", val);
 }
-
+#if 0
 static struct kobj_attribute gen1_interrupt_pin1_enable =
 __ATTR(pin1_enable, 0664, attr_get_gen1_status, attr_set_gen1_status);
 static struct kobj_attribute gen1_interrupt_pin2_enable =
@@ -2564,7 +2564,7 @@ static struct attribute *attributes_mag_interrupt[] = {
 	&gen_mag_z.attr,
 	NULL,
 };
-
+#endif
 static struct attribute *attributes_acc[] = {
 	&poll_attr_acc.attr,
 	&enable_attr_acc.attr,
diff --git a/drivers/amlogic/input/sensor/mir3da_cust.c b/drivers/amlogic/input/sensor/mir3da_cust.c
index 23e8421d11c8..f9639c856298 100755
--- a/drivers/amlogic/input/sensor/mir3da_cust.c
+++ b/drivers/amlogic/input/sensor/mir3da_cust.c
@@ -252,7 +252,7 @@ static long mir3da_misc_ioctl( struct file *file,unsigned int cmd, unsigned long
     int             interval = 0;
     char            bEnable = 0;
     int             z_dir = 0;
-    int             range = 0;
+//    int             range = 0;
     short           xyz[3] = {0};
     MIR_HANDLE      handle = mir_handle;
     
diff --git a/drivers/amlogic/input/touchscreen/gslx680_compatible.c b/drivers/amlogic/input/touchscreen/gslx680_compatible.c
index bc25f9e56b7f..d8306cd91a51 100755
--- a/drivers/amlogic/input/touchscreen/gslx680_compatible.c
+++ b/drivers/amlogic/input/touchscreen/gslx680_compatible.c
@@ -767,7 +767,7 @@ schedule:
 static void gsl_monitor_worker(struct work_struct *work)
 {
 	struct gsl_ts *ts = container_of(to_delayed_work(work), struct gsl_ts, monitor_work);
-	int i;
+//	int i;
 
 #ifdef GSL_NOID_VERSION
 	if (ts->api == NULL) {
diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index 208937bdeaeb..8c3a4c60acde 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -943,7 +943,7 @@ static ssize_t vframe_states_show(struct class *class, struct class_attribute* a
 {
     int ret = 0;
     vframe_states_t states;
-    unsigned long flags;
+//    unsigned long flags;
 	
     if (ionvideo_vf_get_states(&states) == 0) {
         ret += sprintf(buf + ret, "vframe_pool_size=%d\n", states.vf_pool_size);
diff --git a/drivers/amlogic/securitykeys/core.c b/drivers/amlogic/securitykeys/core.c
index c04b1d74af92..8aec61b2fbfb 100644
--- a/drivers/amlogic/securitykeys/core.c
+++ b/drivers/amlogic/securitykeys/core.c
@@ -1299,7 +1299,7 @@ static ssize_t key_name_store(struct device *dev, struct device_attribute *attr,
                              const char *buf, size_t count)
 {
 	aml_key_t * keys;
-	int i,cnt,suffix;
+	int i,cnt;
 	char *name;
 	char *cmd,*oldname=NULL,*newname;
 	if(keys_version == 0){
diff --git a/drivers/amlogic/watchdog/aml_wdt.c b/drivers/amlogic/watchdog/aml_wdt.c
index d4dd0bfa9adc..e7b34b58665e 100644
--- a/drivers/amlogic/watchdog/aml_wdt.c
+++ b/drivers/amlogic/watchdog/aml_wdt.c
@@ -263,15 +263,12 @@ static int aml_wdt_remove(struct platform_device *pdev)
 #ifdef	CONFIG_PM
 static int aml_wdt_suspend(struct platform_device *pdev, pm_message_t state)
 {
-	struct watchdog_device *wdog = platform_get_drvdata(pdev);
 	reset_watchdog();
 	return 0;
 }
 
 static int aml_wdt_resume(struct platform_device *pdev)
 {
-	struct watchdog_device *wdog = platform_get_drvdata(pdev);
-	struct aml_wdt_dev *wdev = watchdog_get_drvdata(wdog);
 	reset_watchdog();
 	return 0;
 }
diff --git a/drivers/media/dvb-core/dvb_frontend.c b/drivers/media/dvb-core/dvb_frontend.c
index 3c77146656b2..fb1a7feea672 100755
--- a/drivers/media/dvb-core/dvb_frontend.c
+++ b/drivers/media/dvb-core/dvb_frontend.c
@@ -575,9 +575,10 @@ static void dvb_frontend_swzigzag(struct dvb_frontend *fe)
 	fe_status_t s = 0;
 	int retval = 0;
 	int time=0;
-	int count=0;
+#if (defined CONFIG_AM_SI2176)
 	int newcount=0;
 	int strength=0;
+#endif
 	struct dvb_frontend_private *fepriv = fe->frontend_priv;
 	struct dtv_frontend_properties *c = &fe->dtv_property_cache, tmp;
 
@@ -636,6 +637,7 @@ static void dvb_frontend_swzigzag(struct dvb_frontend *fe)
 #if (defined CONFIG_AM_M6_DEMOD)
 //dvbc auto qam
 	if(c->modulation== QAM_AUTO){
+	int count=0;
 		while((dvbc_get_status()<=3)&&(count<=20)){
 			msleep(30);
 			if(count==20){
diff --git a/include/linux/amlogic/vout/aml_lcd_common.h b/include/linux/amlogic/vout/aml_lcd_common.h
new file mode 100644
index 000000000000..598a713b5817
--- /dev/null
+++ b/include/linux/amlogic/vout/aml_lcd_common.h
@@ -0,0 +1,13 @@
+#ifndef AML_LCD_COMMON_H
+#define AML_LCD_COMMON_H
+
+static const char* lcd_type_table[]={
+	"MIPI",
+	"LVDS",
+	"eDP",
+	"TTL",
+	"miniLVDS",
+	"invalid",
+};
+
+#endif
diff --git a/include/linux/amlogic/vout/aml_lcd_extern.h b/include/linux/amlogic/vout/aml_lcd_extern.h
index d66aa7c295bd..05762ff77f0e 100755
--- a/include/linux/amlogic/vout/aml_lcd_extern.h
+++ b/include/linux/amlogic/vout/aml_lcd_extern.h
@@ -49,6 +49,7 @@ struct lcd_extern_config_t {
 extern struct aml_lcd_extern_driver_t* aml_lcd_extern_get_driver(void);
 extern int lcd_extern_driver_check(void);
 extern int get_lcd_extern_dt_data(struct device_node* of_node, struct lcd_extern_config_t *pdata);
+extern int remove_lcd_extern(struct lcd_extern_config_t *pdata);
 
 #endif
 
diff --git a/include/linux/amlogic/vout/lcdoutc.h b/include/linux/amlogic/vout/lcdoutc.h
index c06269d84c04..c0457addb67a 100755
--- a/include/linux/amlogic/vout/lcdoutc.h
+++ b/include/linux/amlogic/vout/lcdoutc.h
@@ -86,23 +86,6 @@ typedef enum {
 #define	LCD_POWER_GPIO_OUTPUT_HIGH	1
 #define	LCD_POWER_GPIO_INPUT		2
 
-static const char* lcd_power_type_table[]={
-	"cpu",
-	"pmu",
-	"signal",
-	"init",
-	"null",
-};
-
-static const char* lcd_power_pmu_gpio_table[]={
-	"GPIO0",
-	"GPIO1",
-	"GPIO2",
-	"GPIO3",
-	"GPIO4",
-	"null",
-}; 
-
 typedef enum {
 	LCD_DIGITAL_MIPI = 0,
 	LCD_DIGITAL_LVDS,
@@ -112,15 +95,6 @@ typedef enum {
 	LCD_TYPE_MAX,
 } Lcd_Type_t;
 
-static const char* lcd_type_table[]={
-	"MIPI",
-	"LVDS",
-	"eDP",
-	"TTL",
-	"miniLVDS",
-	"invalid",
-};
-
 typedef struct {
 	char *model_name;
 	u16 h_active;		// Horizontal display area
diff --git a/include/linux/amlogic/vout/vout_notify.h b/include/linux/amlogic/vout/vout_notify.h
index 6c09fdcfb3a1..5ff2dddf08bc 100755
--- a/include/linux/amlogic/vout/vout_notify.h
+++ b/include/linux/amlogic/vout/vout_notify.h
@@ -70,7 +70,7 @@ extern vmode_t validate_vmode(char *);
 extern int vout_suspend(void); 
 extern int vout_resume(void);
 
-extern int get_power_level();
+extern int get_power_level(void);
 
 
 #define VOUT_EVENT_MODE_CHANGE		0x00010000	
diff --git a/sound/soc/aml/m6/aml_audio_hw.c b/sound/soc/aml/m6/aml_audio_hw.c
index 6a8525e165d9..5933441c8cb4 100755
--- a/sound/soc/aml/m6/aml_audio_hw.c
+++ b/sound/soc/aml/m6/aml_audio_hw.c
@@ -372,7 +372,7 @@ int if_audio_in_i2s_enable()
 
 void audio_in_spdif_enable(int flag)
 {
-  int rd = 0, start=0;
+//  int rd = 0, start=0;
 
 	if(flag){
 reset_again:
diff --git a/sound/soc/aml/m6/aml_audio_hw_pcm2bt.c b/sound/soc/aml/m6/aml_audio_hw_pcm2bt.c
index adfa4bda6574..ca1e4414c8c2 100644
--- a/sound/soc/aml/m6/aml_audio_hw_pcm2bt.c
+++ b/sound/soc/aml/m6/aml_audio_hw_pcm2bt.c
@@ -137,7 +137,7 @@ unsigned int pcm_in_set_rd_ptr(unsigned int value)
 
 unsigned int pcm_in_wr_ptr(void)
 {
-	unsigned int writing = READ_MPEG_REG(AUDIN_FIFO1_PTR);
+//	unsigned int writing = READ_MPEG_REG(AUDIN_FIFO1_PTR);
     unsigned int written = 0;
     unsigned int value = 0;
 
diff --git a/sound/soc/aml/m6/aml_m6_wm8960.c b/sound/soc/aml/m6/aml_m6_wm8960.c
index 833e4ad39350..f462aba4df49 100644
--- a/sound/soc/aml/m6/aml_m6_wm8960.c
+++ b/sound/soc/aml/m6/aml_m6_wm8960.c
@@ -478,8 +478,8 @@ static struct device_node * find_avaliable_asoc_device(struct device_node *pnode
 */       
 static int aml_m6_audio_probe(struct platform_device *pdev)
 {
-    struct device_node *np = pdev->dev.of_node;
-    struct device_node *npc;
+//    struct device_node *np = pdev->dev.of_node;
+//    struct device_node *npc;
     struct snd_soc_card *card = &aml_snd_soc_card;
     struct aml_audio_private_data *p_aml_audio;
     int ret = 0;
@@ -549,8 +549,8 @@ err:
 static int aml_m6_audio_remove(struct platform_device *pdev)
 {
     int ret = 0;
-    struct device_node *np = pdev->dev.of_node;
-    struct device_node *npc;
+//    struct device_node *np = pdev->dev.of_node;
+//    struct device_node *npc;
     struct snd_soc_card *card;
     struct aml_audio_private_data *p_aml_audio;
 
diff --git a/sound/soc/aml/m6/aml_pcm.c b/sound/soc/aml/m6/aml_pcm.c
index 0c57912a5d3d..cbfbb922e27a 100755
--- a/sound/soc/aml/m6/aml_pcm.c
+++ b/sound/soc/aml/m6/aml_pcm.c
@@ -82,9 +82,9 @@ unsigned int aml_i2s_alsa_write_addr = 0;
 
 static  unsigned  playback_substream_handle = 0 ;
 /*to keep the pcm status for clockgating*/
-static unsigned clock_gating_status = 0;
-static unsigned clock_gating_playback = 1;
-static unsigned clock_gating_capture = 2;
+//static unsigned clock_gating_status = 0;
+//static unsigned clock_gating_playback = 1;
+//static unsigned clock_gating_capture = 2;
 static int audio_type_info = -1;
 static int audio_sr_info = -1;
 extern unsigned audioin_mode;
@@ -94,14 +94,14 @@ static unsigned trigger_underun = 0;
 static int audio_ch_info = -1;
 
 //static struct rt5631_platform_data *rt5631_snd_pdata = NULL;
-static struct aml_pcm_work_t{
-	struct snd_pcm_substream *substream;
-	struct work_struct aml_codec_workqueue;
-}aml_pcm_work;
+//static struct aml_pcm_work_t{
+//	struct snd_pcm_substream *substream;
+//	struct work_struct aml_codec_workqueue;
+//}aml_pcm_work;
 
 int codec_power=1;
 unsigned int flag=0;
-static int num=0;
+//static int num=0;
 
 static int codec_power_switch(struct snd_pcm_substream *substream, unsigned int status);
 
@@ -863,7 +863,7 @@ static enum hrtimer_restart aml_pcm_hrtimer_callback(struct hrtimer* timer)
   struct snd_pcm_runtime* runtime= substream->runtime;
 
   unsigned int last_ptr, size;
-  unsigned long flag;
+//  unsigned long flag;
   //printk("------------->hrtimer start\n");
   if(s->active == 0){
     hrtimer_forward_now(timer, ns_to_ktime(HRTIMER_PERIOD));
diff --git a/sound/soc/aml/m6/aml_platform.c b/sound/soc/aml/m6/aml_platform.c
index 5dace9fc1b5a..eaef49c58909 100644
--- a/sound/soc/aml/m6/aml_platform.c
+++ b/sound/soc/aml/m6/aml_platform.c
@@ -452,8 +452,8 @@ static int aml_platform_pcm_new(struct snd_soc_pcm_runtime *rtd)
     printk("codec_dai %x \n",	(unsigned)codec_dai);
 	
     struct aml_audio_interface *aud_interface = NULL;
-    struct snd_pcm *pcm =rtd->pcm ;
-    struct snd_soc_card *card = rtd->card;
+//    struct snd_pcm *pcm =rtd->pcm ;
+//    struct snd_soc_card *card = rtd->card;
 	
 	int ret = 0;
 
diff --git a/sound/soc/aml/m6/aml_platform_pcm2bt.c b/sound/soc/aml/m6/aml_platform_pcm2bt.c
index 39475697d9ba..2aec2b81ddf6 100644
--- a/sound/soc/aml/m6/aml_platform_pcm2bt.c
+++ b/sound/soc/aml/m6/aml_platform_pcm2bt.c
@@ -576,7 +576,7 @@ static int aml_pcm2bt_new(struct snd_soc_pcm_runtime *rtd)
 	int ret = 0;
        struct snd_soc_card *card = rtd->card;
        struct snd_pcm *pcm =rtd->pcm ;  
-       struct snd_soc_dai *dai =rtd->cpu_dai ;  	   
+//       struct snd_soc_dai *dai =rtd->cpu_dai ;  	   
        pcm_debug("enter %s dai->name: %s dai->id: %d\n", __FUNCTION__, dai->name, dai->id);
     
 	if (!card->dev->dma_mask)
diff --git a/sound/soc/aml/m8/aml_m8.c b/sound/soc/aml/m8/aml_m8.c
index bb9f93640734..200b9c6d0c87 100755
--- a/sound/soc/aml/m8/aml_m8.c
+++ b/sound/soc/aml/m8/aml_m8.c
@@ -338,7 +338,6 @@ static int aml_audio_get_i2s_mute(struct snd_kcontrol *kcontrol,
 }
 
 
-static bool spdif_mute_flag;
 
 static int aml_audio_set_spdif_mute(struct snd_kcontrol *kcontrol,
     struct snd_ctl_elem_value *ucontrol)
-- 
2.19.0

